From 503809478bf13369a2fdb455a075f5268d17e6f0 Mon Sep 17 00:00:00 2001
From: Marek 'marx' Grac <mgrac@redhat.com>
Date: Wed, 13 Nov 2013 14:47:27 +0100
Subject: [PATCH 4/5] Resolves: rhbz#1029755 - allow online extension of ext3
 LV

Previously, only offline extension of ext3 filesystem was allowed.

Patch by: Siddharth Sharma
---
 src/Filesystem.py | 26 ++------------------------
 1 file changed, 2 insertions(+), 24 deletions(-)

diff --git a/src/Filesystem.py b/src/Filesystem.py
index f72930d..c55a144 100644
--- a/src/Filesystem.py
+++ b/src/Filesystem.py
@@ -272,7 +272,7 @@ class ext3(Filesystem):
         creatable = self.check_path('/sbin/mkfs.ext3')
         mountable = self.check_mountable('ext3', 'ext3')
         resize_offline = self.check_paths(['/sbin/e2fsck', '/sbin/resize2fs'])
-        extend_online, dummy = self.__extend_online_cmd()
+        extend_online = self.check_path('/sbin/resize2fs')
         
         Filesystem.__init__(self, 'Ext3', creatable, True, mountable, 
                             extend_online, resize_offline, False, resize_offline,
@@ -302,35 +302,13 @@ class ext3(Filesystem):
             raise CommandError('FATAL', FSCREATE_FAILURE % (cmdstr,e))
     
     def extend_online(self, dev_path):
-        dummy, cmd = self.__extend_online_cmd()
+        cmd = '/sbin/resize2fs'
         args = [cmd, dev_path]
         cmdstr = ' '.join(args)
         msg = RESIZING_FS % (self.name)
         o, e, s = execWithCaptureErrorStatusProgress(cmd, args, msg)
         if s != 0:
             raise CommandError('FATAL', FSRESIZE_FAILURE % (cmdstr, e))
-    def __extend_online_cmd(self):
-        global resize2fs_v139plus;
-        
-        # if ext2online is present, use it
-        cmd = '/usr/sbin/ext2online'
-        if self.check_path(cmd):
-            return (True, cmd)
-        # resize2fs 1.39 and above are also able to online extend ext3
-        try:
-            cmd = '/sbin/resize2fs'
-            if resize2fs_v139plus != None:
-                return (True, cmd)
-            
-            o, e, s = execWithCaptureErrorStatus(cmd, [cmd])
-            if (s == 0 or s == 1) and float(e.strip().split()[1]) >= 1.39:
-                resize2fs_v139plus = True
-                return (True, cmd)
-            else:
-                resize2fs_v139plus = False
-        except:
-            pass
-        return (False, None)
     
     def reduce_online(self, dev_path, new_size_bytes):
         # not supported
-- 
2.1.4

